opal/datatype/opal_datatype_copy.c

```diff
@@ -67,78 +55,104 @@ static size_t opal_datatype_memop_block_size = 128 * 1024;
         }                                                                                   \
     } while (0)
 
-/**
- * Overlapping memory regions
- */
+static inline void opal_datatype_accelerator_memcpy_impl(void *dest, const void *src, size_t size, int dev_id, 
+                                                          opal_accelerator_transfer_type_t type)
+{
+    int res = opal_accelerator.get_device(&dev_id);
+    if (OPAL_SUCCESS != res) {
+        /**
+         * Attempt the copy with 0 as dev_id. If we fail the copy then we'll abort anyway.
+         */
+        dev_id = 0;
+    }
+    res = opal_accelerator.memcpy(dev_id, dest, src, size, type);
+    if (OPAL_SUCCESS != res) {
```

Comment:
I'm thinking it's not really that useful. I added that a while ago because of the way ZE memcpy works...but there's probably a way for the implementation to handle it. Since the libfabric API had a dev_id, I figured that it was there mainly for ZE

Reply:
Added it to give more flexibility in the future, but also allowed -1 (no device id) as a valid input